src/core/public/chrome/chrome_service.tsx

```diff
@@ -94,11 +89,38 @@ export class ChromeService {
     injectedMetadata,
     notifications,
   }: StartDeps): Promise<InternalChromeStart> {
-    const FORCE_HIDDEN = isEmbedParamInHash();
+    // Start off the chrome service hidden if "embed" is in the hash query string.
+    const isEmbedded = 'embed' in parse(location.hash.slice(1), true).query;
+
+    /**
+     * These observables allow consumers to toggle the chrome visibility via either:
+     *   1. Using setIsVisible() to trigger the next chromeHidden$
+     *   2. Setting `chromeless` when registering an application, which will
+     *      reset the visibility whenever the next application is mounted
+     *   3. Having "embed" in the query string
+     */
+    const chromeHidden$ = new BehaviorSubject(isEmbedded);
```

Comment:
nit: maybe we should rename this to `manualHidden$` so it's less ambiguous with `isVisible$`

Reply:
I renamed this to `toggleHidden$` to convey that this can be changed and a little more action-descriptive.